app/src/lib/git/core.ts

```diff
@@ -182,9 +182,14 @@ function getDescriptionForError(error: DugiteError): string {
     case DugiteError.SSHAuthenticationFailed:
     case DugiteError.SSHPermissionDenied:
     case DugiteError.HTTPSAuthenticationFailed:
-      return `Authentication failed. You may not have permission to access the repository or the repository may have been archived. Open ${
-        __DARWIN__ ? 'preferences' : 'options'
-      } and verify that you're signed in with an account that has permission to access this repository.`
+      return `Authentication failed. Some common reasons include:
+- You are not logged in to your account: see ${
+        __DARWIN__ ? 'Preferences' : 'Options'
+      }
+- You may need to logout and login to refresh your token.
+- You do not have permission to access this repository.
+- The repository is archived. Check the repository settings on GitHub.com
+- If you use SSH, check the validity of your key.`
```

Comment:
:art: I'm not sure how much knowledge you think users will have if they get stuck but also use SSH authentication, so here's a suggestion to add some specific things to look for:

```suggestion
- If you use SSH authentication, check your key is added to the SSH agent and associated with your account.`
```